File: CodingSamples/DotNet/Advanced/Networking/TcpClientTest/DemoApp2/Models/ShopModel.cs
namespace DemoApp.Models;

public class ShopModel(string remote)
{
    public async ValueTask<ItemInfo> FetchItemInfoAsync(string name)
    {
        using var client = new HttpClient 
        {
            BaseAddress = new Uri($"http://{remote}/shop/")
        };
        var response = await client.GetAsync(name);
        if(response.IsSuccessStatusCode)
        {
            var message = await response.Content.ReadAsStringAsync();
            return ItemInfo.Parse(message);
        }
        return default; 
    }
}

File: CodingSamples/Java/Advanced/Networking/TCPClientTest/demo2/src/main/java/app/models/ShopModel.java
package app.models;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;

public class ShopModel {
    
    private String server;

    public ShopModel(String remote) {
        server = remote;
    }

    public ItemInfo fetchItemInfo(String name) throws IOException, InterruptedException {
        var url = URI.create("http://" + server + "/shop/" + name);
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder(url)
            .GET()
            .header("User-Agent", "TCPClientTest demo/2.0")
            .build();
        var response = client.send(request, ItemInfo.bodyHandler());
        return response.body();

    }
}
